root.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"net"
+	"net/http"
+
+	"github.com/beanstalkg/beanstalkg/architecture"
+	"github.com/beanstalkg/beanstalkg/operation"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var serverCmd = &cobra.Command{
+	Use:   "beanstalkg",
+	Short: "Beanstalkg is a go implementation of beanstalkd - A fast, general-purpose work queue",
+	Run:   startServer,
+}
+
+func startServer(cmd *cobra.Command, args []string) {
+	port := viper.GetString("port")
+	debugMode := viper.GetBool("debug_mode")
+
+	initLogging(debugMode)
+	go func() {
+		log.Info(http.ListenAndServe("localhost:6060", nil))
+	}()
+
+	service := ":" + port
+	tcpAddr, err := net.ResolveTCPAddr("tcp4", service)
+	checkError(err)
+	listener, err := net.ListenTCP("tcp", tcpAddr)
+	checkError(err)
+	stop := make(chan bool)
+
+	tubeRegister := make(chan architecture.Command)
+	// use this tube to send the channels for each individual tube to the clients when the do 'use' command
+	useTubeConnectionReceiver := make(chan chan architecture.Command)
+	watchedTubeConnectionsReceiver := make(chan chan architecture.Command)
+	operation.NewTubeRegister(tubeRegister, useTubeConnectionReceiver, watchedTubeConnectionsReceiver, stop)
+	log.Info("BEANSTALKG listening on: ", port)
```

Comment:
The best practices you linked in #66 recommends embedding the logger in a struct so it's not a free variable.  That would fix the logging issues we discussed earlier.

Would you object to me handling all logging issues in a separate PR?